content/filterx/function-reference.md

```diff
@@ -0,0 +1,306 @@
+---
+title: Filterx function reference
+linkTitle: Functions
+weight: 4800
+---
+
+<!-- This file is under the copyright of Axoflow, and licensed under Apache License 2.0, except for using the Axoflow and AxoSyslog trademarks. -->
+
+## cache_json_file
+
+Usage: `cache_json_file("/path/to/file.json")`
+
```

Comment:
Load the contents of a JSON file in an efficient manner. You can use this to lookup contextual information. For example your file.json contains this: `{"nginx": "web", "httpd": "web", "apache": "web", "mysql": "db", "postgresql": "db"}`, you can use the following filterx expression to route "web" traffic only:

```filterx {
declare apps = cache_json_file("/path/to/file.json");
$app = apps[$PROGRAM] ?? "unknown";
$app == "web";  # drop everything that's not a web server log
}`

Currently the JSON file's contents are updated when the <product> config is reloaded. (We have plans to make it automatic)